cluster/swarm/cluster.go

```diff
@@ -371,12 +371,52 @@ func (c *Cluster) Container(IDOrName string) *cluster.Container {
 
 	c.RLock()
 	defer c.RUnlock()
-	for _, n := range c.engines {
-		if container := n.Container(IDOrName); container != nil {
+
+	containers := c.Containers()
+
+	// Match exact or short Container ID.
+	for _, container := range containers {
+		if container.Id == IDOrName || stringid.TruncateID(container.Id) == IDOrName {
+			return container
+		}
+	}
+
+	// Match exact Swarm ID.
+	for _, container := range containers {
+		if swarmID := container.Config.SwarmID(); swarmID == IDOrName || stringid.TruncateID(swarmID) == IDOrName {
 			return container
 		}
 	}
 
+	// Match name, /name or engine/name.
```

Comment:
Maybe do the same here?

Since people can create containers outside of swarm, we may end up with a name collision